include/qore/intern/QC_DebugProgram.h

```diff
@@ -45,21 +45,34 @@ class QoreDebugProgramWithQoreObject: public QoreDebugProgram {
 private:
    QoreObject* qo;
 
-   DLLLOCAL void callMethod(const char* name, QoreProgram *pgm, int paramCount, QoreValue* params, DebugRunStateEnum &rs, ExceptionSink* xsink, ExceptionSink &xsink2) {
-      ReferenceHolder<QoreListNode> args(new QoreListNode, &xsink2);
+   DLLLOCAL void callMethod(const char* name, QoreProgram *pgm, int paramCount, QoreValue* params, DebugRunStateEnum &rs, const AbstractStatement* &rts, ExceptionSink* xsink, ExceptionSink &xsink2) {
+      ReferenceHolder<QoreListNode> args(new QoreListNode(), &xsink2);
       args->push(QoreProgram::getQoreObject(pgm), nullptr);
       for (int i=0; i<paramCount; i++) {
          //printd(5, "QoreDebugProgramWithCoreObject::callMethod(%s) this: %p, param: %d/%d, type: %s\n", name, this, i, paramCount, params[i]?params[i]->getTypeName():"n/a");
          args->push(params[i], nullptr);
       }
       // LocalVar will sanitize and discard non-node values so we cannot use the ReferenceHolder
-      ReferenceArgumentHelper rah(rs, &xsink2);
-      args->push(rah.getArg(), nullptr); // caller owns ref
-      printd(5, "QoreDebugProgramWithCoreObject::callMethod(%s) this: %p, pgm: %p, param#: %d, rs: %d, xsink2: %d\n", name, this, pgm, paramCount, rs, xsink2.isEvent());
-      qo->evalMethod(name, *args, &xsink2).discard(&xsink2);
-      QoreValue v(rah.getOutputValue());
-      rs = (DebugRunStateEnum) v.getAsBigInt();
-      v.discard(&xsink2);
+      ReferenceArgumentHelper rsah(rs, &xsink2);
+      args->push(rsah.getArg(), nullptr); // caller owns ref
+      int sid = rts ? pgm->getStatementId(rts) : 0;
+      ReferenceArgumentHelper rtsah(sid, &xsink2);
+      args->push(rtsah.getArg(), nullptr); // caller owns ref
+      printd(5, "QoreDebugProgramWithCoreObject::callMethod(%s) this: %p, pgm: %p, param#: %d, rs: %d, rts: %d, xsink2: %d\n", name, this, pgm, paramCount, rs, sid, xsink2.isEvent());
+      qo->evalMethod(name, *args, &xsink2);
+      QoreValue rsv(rsah.getOutputValue());
+      rs = (DebugRunStateEnum) rsv.getAsBigInt();
+      rsv.discard(&xsink2);
+
+      QoreValue rtsv(rtsah.getOutputValue());
+      sid = rtsv.getAsBigInt();
+      rtsv.discard(&xsink2);
+      if (sid <= 0) {
+        rts = nullptr;
+      } else {
+        rts = pgm->resolveStatementId(sid);
+      }
```

Comment:
can you please come up with better variable names? It's a bit confusing to have `rs` `rsv` `rts` `rtsv` `rsah` `rtsah`

Reply:
also please use 4 spaces for indentation please - no tabs

Reply:
This naming convention style is widely used in Qore C++ code. These variables are temporary so it is equivalent of `i` for counters or so. 

There is no tab but wrong 2 space indent as result of mixing 3 and 4 space indentation (C vs. Qore).

Reply:
I know it's temporary variables, but even so we can come up with better names for them. 

> This naming convention style is widely used in Qor
true, and I would like to change it for better readability and maintainability

Le'ts use 4 spaces everywhere (c++, c, qore, qpp, qorus, ....)

Reply:
The code is there some time. So let's create new issue for "better readability and maintainability" and unify all variable names in all sources. If you have an idea about these temporary var names so feel free and put suggestions and if I agree that is more readable I can update it. My fantasy in this case is not sufficient.

So reformat all sources to 4 space indentation. It is not in scope of this pull request.